### PR TITLE
(fix) remove order proposals with sizes below the specified order_amount

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue: #1893
- Related Clubhouse Story:

**A description of the changes proposed in the pull request**:

### CODE REVIEW

File: hummingbot/strategy/pure_market_making/pure_market_making.pyx [hunk 1]
```diff
@@ -677,16 +677,17 @@ cdef class PureMarketMakingStrategy(StrategyBase):
                                         buy.size, buy.price)
             quote_size = buy.size * buy.price * (Decimal(1) + buy_fees.percent)
             if quote_balance < quote_size_total + quote_size:
-                quote_size = quote_balance - quote_size_total
-                base_size = quote_size / buy.price * (Decimal(1) - buy_fees.percent)
-                buy.size = market.c_quantize_order_amount(self.trading_pair, base_size, buy.price)
+                self.logger().info(f"Insufficient balance: Quote order size {quote_size} is above available quote balance {quote_balance - quote_size_total}.")
```
Comment (Nullably) [inline hummingbot/strategy/pure_market_making/pure_market_making.pyx:680]: I think it'd be more informative for the message to be more like
`Insufficient balance: Buy order (price: XXX, size: xxx) is omitted, {self._quote_asset} available balance: XXX`

File: hummingbot/strategy/pure_market_making/pure_market_making.pyx [hunk 2]
```diff
@@ -677,16 +677,17 @@ cdef class PureMarketMakingStrategy(StrategyBase):
...
+                quote_size = s_decimal_zero
+                buy.size = s_decimal_zero
             quote_size_total += quote_size
         proposal.buys = [o for o in proposal.buys if o.size > 0]
         for sell in proposal.sells:
             base_size = sell.size
             if base_balance < base_size_total + base_size:
-                base_size = base_balance - base_size_total
-                sell.size = market.c_quantize_order_amount(self.trading_pair, base_size, sell.price)
+                self.logger().info(f"Insufficient balance: Base order size {base_size} is above Available base balance {base_balance - base_size_total}.")
```
Comment (Nullably) [inline hummingbot/strategy/pure_market_making/pure_market_making.pyx:688]: I think it'd be more informative for the message to be more like
`Insufficient balance: Sell order (price: XXX, size: xxx) is omitted, {self._base_asset} available balance: XXX`

File: hummingbot/strategy/pure_market_making/pure_market_making.pyx (hunk 1 above)
Comment (vic-en) [reply hummingbot/strategy/pure_market_making/pure_market_making.pyx]: Done.

File: hummingbot/strategy/pure_market_making/pure_market_making.pyx (hunk 2 above)
Comment (vic-en) [reply hummingbot/strategy/pure_market_making/pure_market_making.pyx]: Done.

**Comment by mifeng** [comment]
test this please

**Comment by carlolm** [comment]
image

@vic-en here's the test failure log

**Pull request merged**